### PR TITLE
[MM-51233] Updates for Trial Requests

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary
This PR adds a new Trial Request Form, leveraged from all Channels entrypoints into the Trial Request flow. This new form asks for new information including:

Name
Email (that is validated against the CWS to be a business email)
Company Name
Company Size
Company Country

These fields are passed to the backend and the backend sends them along with its trial request to the CWS. The CWS does not yet support these new fields, but should consume the trial requests with the supported existing fields just fine. 

Explicit support for embargoed entities will be covered in a future PR, under ticket: https://mattermost.atlassian.net/browse/MM-51470

#### Screenshots
The email field is pre-filled based on the current user's email address
image

Business email is validated on blur
image

Cannot submit without valid business email 
image

Filled form
image

Clicking Start Trial opens the trial benefits modal on success
image

Failure to activate trial request results in modal:
image

This is slightly different from the designs as it allowed me to re-use some existing components as well as allow the following UX: Clicking Try Again returns you to the Start Trial modal with the same information you entered originally. For now, this error modal will be for _all errors_ including export compliance errors, and air gapped instances (to be fixed in a future iteration)
image

#### Ticket Link
This PR encapsulates a few different tickets:
https://mattermost.atlassian.net/browse/MM-51348
https://mattermost.atlassian.net/browse/MM-51233
https://mattermost.atlassian.net/browse/MM-51424

#### Release Note

```release-note
Updates to trial request forms to allow for a more tailored trial experience
```

### CODE REVIEW

**Comment by nickmisasi** [comment]
Marked as draft since this is based off the mono-repo branch. Should still be good to review.

**Review by pvev — APPROVED** [review]
LGTM! just a couple of nits. Thanks @nickmisasi :shipit:

File: webapp/channels/src/components/start_trial_form_modal/index.tsx
Comment (neallred) [inline webapp/channels/src/components/start_trial_form_modal/index.tsx:89]: This works since we only want run once behavior, but breaks the "rules of hooks" since it doesn't include the email in the depencency array. To be totally conformant with that, I'd recommend adding a "didOnce" boolean state flag

```ts
const [email, setEmail] = useState(currentUser.Email);
const [didOnce, setDidOnce] = useState(false);

    useEffect(() => {
        if (didOnce) {
            return;
        }
        setDidOnce(true);
        trackEvent(TELEMETRY_CATEGORIES.SELF_HOSTED_START_TRIAL_MODAL, 'form_opened');
        if (email) {
                    handleValidateBusinessEmail(email);
        }
    }, [email, doOnce]);
```

File: webapp/channels/src/components/start_trial_form_modal/index.tsx
Comment (neallred) [inline webapp/channels/src/components/start_trial_form_modal/index.tsx:84]: Nit: type assertion not needed.

File: webapp/channels/src/components/start_trial_form_modal/index.tsx
Comment (neallred) [reply webapp/channels/src/components/start_trial_form_modal/index.tsx:23]: Not sure if I'm just saying the same thing differently, but looks like mattermost-redux and actions should be grouped above with the other mattermost-redux and actions imports.

**Review by neallred — APPROVED** [review]
LGTM, thanks @nickmisasi ! There were some nits, but nothing blocking.

This is probably already known and over-communicating, but heads up that we won't want to merge this into `mono-repo`, and will instead want to wait until `mono-repo` is merged into `master` : https://community.mattermost.com/core/pl/k4bbzmq8wbrqugpyxz3xj9dimo

File: webapp/channels/src/components/start_trial_form_modal/failure_modal.tsx
Comment (nickmisasi) [reply webapp/channels/src/components/start_trial_form_modal/failure_modal.tsx:22]: TIL - nice

File: webapp/channels/src/components/admin_console/billing/delete_workspace/result_modal.tsx
Comment (nickmisasi) [reply webapp/channels/src/components/admin_console/billing/delete_workspace/result_modal.tsx:55]: I'm having some issues getting that component to work with the mono repo. It could just be my local, but not sure

**Pull request closed without merging**
